src/core/lib/gprpp/ref_counted.h

```diff
@@ -355,12 +355,12 @@ class RefCounted : public Impl {
   // friend of this class.
   void Unref() const {
     if (GPR_UNLIKELY(refs_.Unref())) {
-      unref_behavior_(static_cast<const Child*>(this));
+      unref_behavior_(const_cast<Child*>(static_cast<const Child*>(this)));
```

Comment:
Why is this needed?  I don't think any of the values for `unref_behvaior_` require the object to be non-`const`.  And if the object is actually `const`, then this seems unsafe.

Update: Reading further, it looks like this is because `Arena::Destroy()` is non-`const`.  Could we avoid that by simply moving the code from `Destroy()` into the `Arena` dtor, and using the `UnrefCallDtor` unref behavior?

Reply:
that's nicer yeah... still end up with a const cast, but it's passing the arena ptr to gpr_free_aligned and that seems fine